src/index.ml

```diff
@@ -933,11 +938,12 @@ struct
     ignore (replace' ?hook:None ?overcommit t key value : _ async option)
 
   let replace_with_timer ?sampling_interval t key value =
-    if sampling_interval <> None then Stats.start_replace ();
-    replace t key value;
     match sampling_interval with
-    | None -> ()
-    | Some sampling_interval -> Stats.end_replace ~sampling_interval
+    | None -> replace t key value (* XXX(craigfe): why not log duration here? *)
```

Comment:
it's just a convention we made for when `sampling_interval` is not set https://github.com/mirage/index/blob/master/src/index_intf.ml#L255

Reply:
OK, thanks!